src/cljs/main/broadfcui/page/profile.cljs

```diff
@@ -123,62 +124,51 @@
+                      (this :render-field :programLocationCity "City")
+                      (this :render-field :programLocationState "State/Province")
+                      (this :render-field :programLocationCountry "Country"))
             (common/clear-both)
             (when-not (:new-registration? props)
-              [:div {} [NihLink (select-keys props [:nih-token])]])]
+              [NihLink (select-keys props [:nih-token])])]
            :else (spinner "Loading User Profile...")))
    :render-radio-field
    (fn [{:keys [state]} key value]
-     [:div {:style {:float "left" :margin "0 1em 0.5em 0" :padding "0.5em 0"}}
-      [:label {}
-       [:input {:type "radio" :value value :name key
-                :checked (= (@user-info/saved-user-profile key) value)
-                :onChange #(swap! state assoc-in [:values key] value)}]
-       value]])
-   :render-nested-field
-   (fn [{:keys [state]} key label required]
-     [:div {:style {:float "left" :marginBottom "0.5em" :marginTop "0.5em"}}
-      [:label {}
-       [:div {:style {:fontSize "88%"}} label]]
-      [input/TextField {:style {:marginRight "1em" :width 200}
-                        :data-test-id key
-                        :defaultValue (@user-info/saved-user-profile key)
-                        :ref (name key)
-                        :predicates [(when required (input/nonempty label))]
-                        :onChange #(swap! state assoc-in [:values key] (-> % .-target .-value))}]])
+     [:label {:style {:margin "0 1em 0.5em 0" :padding "0.5em 0"}}
+      [:input {:type "radio" :value value :name key
+               :checked (= value (get-in @state [:values key] (@user-info/saved-user-profile key)))
+               :onChange #(swap! state assoc-in [:values key] value)}]
+      value])
    :render-field
-   (fn [{:keys [state]} key label required valid-email-or-empty]
-     [:div {:style {:clear "both" :margin "0.5em 0"}}
-      [:label {}
-       (style/create-form-label label)
-       [input/TextField {:style {:width 200}
-                         :data-test-id key
-                         :defaultValue (@user-info/saved-user-profile key)
-                         :ref (name key)
-                         :placeholder (when valid-email-or-empty
-                                        (-> @utils/auth2-atom
-                                            (.-currentUser) (.get) (.getBasicProfile) (.getEmail)))
-                         :predicates [(when required (input/nonempty label))
-                                      (when valid-email-or-empty (input/valid-email-or-empty label))]
-                         :onChange #(swap! state assoc-in [:values key] (-> % .-target .-value))}]]])
+   (fn [{:keys [state]} key label & flags]
+     (let [flag-set (set flags)
+           required? (not (flag-set :optional))
+           email? (flag-set :email)]
+       [:div {:style {:margin "0.5em 1em 0.5em 0"}}
+        (style/create-form-label label)
```

Comment:
Labels should contain the associated element, so that clicking the one gives keyboard focus to the other.